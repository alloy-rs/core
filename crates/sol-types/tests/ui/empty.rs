use alloy_sol_types::sol;

sol! {}

sol! {
    struct EmptyStruct {}
}

sol! {
    enum EmptyEnum {}
}

// OK
sol! {
    contract EmptyContract {}
}

sol! {
    error EmptyError();
}

sol! {
    event EmptyEvent();
}

sol! {
    function emptyFunction();
}

fn main() {}
