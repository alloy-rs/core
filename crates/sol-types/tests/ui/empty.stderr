error: unexpected end of input, expected one of: `constructor`, `function`, `fallback`, `receive`, `modifier`, `struct`, `event`, `error`, `abstract`, `contract`, `interface`, `library`, `enum`, `type`, `pragma`, `import`, `using`, parentheses, `tuple`, `function`, `mapping`, identifier
 --> tests/ui/empty.rs:3:1
  |
3 | sol! {}
  | ^^^^^^^
  |
  = note: this error originates in the macro `sol` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, defining empty structs is disallowed
 --> tests/ui/empty.rs:6:25
  |
6 |     struct EmptyStruct {}
  |                         ^
